client/consensus/grandpa/utils.go

```diff
@@ -0,0 +1,21 @@
+// Copyright 2023 ChainSafe Systems (ON)
+// SPDX-License-Identifier: LGPL-3.0-only
+
+package grandpa
+
+import "fmt"
+
+// SearchKey TODO for reviewer, this can be done by slices.BinarySearch however since it is a tuple being compared,
+// it's unclear to me what value to sort on
+func SearchKey(key Key, changes []PendingChange) (int, error) {
```

Comment:
this is wrong, it needs to return the index of where the key (in this case the block number) should be placed in the slice if it's not found.  I use it in grandpa [here](https://github.com/ChainSafe/gossamer/blob/332b301f0c86ae8fe53f33a8acb0cba868e2a356/pkg/finality-grandpa/dummy_chain_test.go#L104).  You need to ensure the slice is sorted.